src/libcore/hint.rs

```diff
@@ -104,11 +104,28 @@ pub fn spin_loop() {
     }
 }
 
-/// A function that is opaque to the optimizer, to allow benchmarks to
-/// pretend to use outputs to assist in avoiding dead-code
-/// elimination.
+/// An identity function whose argument is opaque to the optimizer. This is useful for (e.g.)
+/// benchmarking, where you want to avoid things like dead-code elimination and constant
```

Comment:
This should probably be part of the body of the documentation and not of the "brief" string. 

I would leave this as "This is useful for, e.g., benchmarking, to prevent benchmarks from being optimized away". There are no guarantees whatsoever about which optimizations this intends to inhibit.